[mpir] Build error

**Host Environment**
- OS: Windows
- Compiler:     
    vcpkg-tool version: 2022-02-24-7efe7f2b04912a7d674f86a406f9f2711c0221c6
    vcpkg-scripts version: 1b1ae50e1 2022-02-25 (3 days ago)

**To Reproduce**
Steps to reproduce the behavior:
`.\vcpkg\vcpkg install mpir:x86-windows-static`

**Failure logs** 
Computing installation plan...
The following packages will be built and installed:
    mpir[core]:x86-windows-static -> 3.0.0#11
Detecting compiler hash for triplet x86-windows-static...
Restored 0 packages from C:\Users\Administrator\AppData\Local\vcpkg\archives in 123.2 us. Use --debug to see more details.
Starting package 1/1: mpir:x86-windows-static
Building package mpir[core]:x86-windows-static...
-- Using community triplet x86-windows-static. This triplet configuration is not guaranteed to succeed.
-- [COMMUNITY] Loading triplet configuration from: C:\Users\Administrator\vcpkg\triplets\community\x86-windows-static.cmake
-- Downloading https://github.com/wbhart/mpir/archive/mpir-3.0.0.tar.gz -> wbhart-mpir-mpir-3.0.0.tar.gz...
[DEBUG] Feature flag 'binarycaching' unset
[DEBUG] Feature flag 'manifests' = off
[DEBUG] Feature flag 'compilertracking' unset
[DEBUG] Feature flag 'registries' unset
[DEBUG] Feature flag 'versions' unset
[DEBUG] Downloading https://github.com/wbhart/mpir/archive/mpir-3.0.0.tar.gz
Error: Failed to download from mirror set:
https://github.com/wbhart/mpir/archive/mpir-3.0.0.tar.gz: failed: **status code 300**

[DEBUG] D:\a\_work\1\s\src\vcpkg\base\downloads.cpp(705)
[DEBUG] Time in subprocesses: 0 us
[DEBUG] Time in parsing JSON: 2 us
[DEBUG] Time in JSON reader: 0 us
[DEBUG] Time in filesystem: 3809 us
[DEBUG] Time in loading ports: 0 us
[DEBUG] Exiting after 847.6 ms (630145 us)

CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:84 (message):

      Failed to download file with error: 1
      If you use a proxy, please check your proxy setting. Possible causes are:

      1. You are actually using an HTTP proxy, but setting HTTPS_PROXY variable
         to `https://address:port`. This is not correct, because `https://` prefix
         claims the proxy is an HTTPS proxy, while your proxy (v2ray, shadowsocksr
         , etc..) is an HTTP proxy. Try setting `http://address:port` to both
         HTTP_PROXY and HTTPS_PROXY instead.

      2. You are using Fiddler. Currently a bug (https://github.com/microsoft/vcpkg/issues/17752)
         will set HTTPS_PROXY to `https://fiddler_address:port` which lead to problem 1 above.
         Workaround is open Windows 10 Settings App, and search for Proxy Configuration page,
         Change `http=address:port;https=address:port` to `address`, and fill the port number.

      3. Your proxy's remote server is out of service.

      In future vcpkg releases, if you are using Windows, you no longer need to set
      HTTP(S)_PROXY environment variables. Vcpkg will simply apply Windows IE Proxy
      Settings set by your proxy software. See (https://github.com/microsoft/vcpkg-tool/pull/49)
      and (https://github.com/microsoft/vcpkg-tool/pull/77)

      Otherwise, please submit an issue at https://github.com/Microsoft/vcpkg/issues

Call Stack (most recent call first):
  scripts/cmake/vcpkg_download_distfile.cmake:309 (z_vcpkg_download_distfile_show_proxy_and_fail)
  scripts/cmake/vcpkg_from_github.cmake:175 (vcpkg_download_distfile)
  ports/mpir/portfile.cmake:59 (vcpkg_from_github)
  scripts/ports.cmake:145 (include)


Error: Building package mpir:x86-windows-static failed with: BUILD_FAILED

**Additional context**
I opened this url https://github.com/wbhart/mpir/archive/mpir-3.0.0.tar.gz in Google Chrome, it turns out `the given path has multiple possibilities: #<Git::Ref:0x00007ff572232868>, #<Git::Ref:0x00007ff5722316e8>`, so I guess this url lacks of some params?

--- a/ports/mpir/portfile.cmake
+++ b/ports/mpir/portfile.cmake
@@ -59,8 +59,8 @@ elseif(VCPKG_TARGET_IS_WINDOWS)
     vcpkg_from_github(
         OUT_SOURCE_PATH SOURCE_PATH
         REPO wbhart/mpir
-        REF mpir-${MPIR_VERSION}
-        SHA512 7d37f60645c533a6638dde5d9c48f5535022fa0ea02bafd5b714649c70814e88c5e5e3b0bef4c5a749aaf8772531de89c331716ee00ba1c2f9521c2cc8f3c61b
+        REF cdd444aedfcbb190f00328526ef278428702d56e # 3.0.0
+        SHA512 f46e45bdba27c9f89953ba23186b694486fd3010bd370ea2de71a4649a2816e716a6520c9baa96936f1884437ef03f92b21c0b1fb5b757beba5a05fed30b2bfc
         HEAD_REF master
         PATCHES enable-runtimelibrary-toggle.patch
     )

--- a/ports/mpir/portfile.cmake
+++ b/ports/mpir/portfile.cmake
@@ -6,31 +6,25 @@ if(VCPKG_CRT_LINKAGE STREQUAL "static" AND VCPKG_LIBRARY_LINKAGE STREQUAL "dynam
     message(FATAL_ERROR "${PORT} currently can only be built using the dynamic CRT when building DLLs")
 endif()
 
-set(MPIR_VERSION 3.0.0)
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO wbhart/mpir
+    REF cdd444aedfcbb190f00328526ef278428702d56e # 3.0.0
+    SHA512 f46e45bdba27c9f89953ba23186b694486fd3010bd370ea2de71a4649a2816e716a6520c9baa96936f1884437ef03f92b21c0b1fb5b757beba5a05fed30b2bfc
+    HEAD_REF master
+    PATCHES enable-runtimelibrary-toggle.patch
+)
 
 if(VCPKG_TARGET_IS_LINUX OR VCPKG_TARGET_IS_OSX)
-    vcpkg_download_distfile(
-        ARCHIVE
-        URLS "http://mpir.org/mpir-${MPIR_VERSION}.tar.bz2"
-        FILENAME mpir-${MPIR_VERSION}.tar.bz2
-        SHA512 c735105db8b86db739fd915bf16064e6bc82d0565ad8858059e4e93f62c9d72d9a1c02a5ca9859b184346a8dc64fa714d4d61404cff1e405dc548cbd54d0a88e
-    )
-
-    vcpkg_extract_source_archive_ex(
-        OUT_SOURCE_PATH SOURCE_PATH
-        ARCHIVE ${ARCHIVE}
-        REF ${MPIR_VERSION}
-    )
-
     vcpkg_find_acquire_program(YASM)
 
     if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
-        set(SHARED_STATIC --enable-static --disable-shared)
+        set(SHARED_STATIC "--enable-static --disable-shared")
     else()
-        set(SHARED_STATIC --disable-static --enable-shared)
+        set(SHARED_STATIC "--disable-static --enable-shared")
     endif()
 
-    set(OPTIONS --disable-silent-rules --enable-gmpcompat --enable-cxx ${SHARED_STATIC})
+    set(OPTIONS "--disable-silent-rules --enable-gmpcompat --enable-cxx ${SHARED_STATIC}")
 
     string(APPEND VCPKG_C_FLAGS " -Wno-implicit-function-declaration")
     string(APPEND VCPKG_CXX_FLAGS " -Wno-implicit-function-declaration")
@@ -41,7 +35,7 @@ if(VCPKG_TARGET_IS_LINUX OR VCPKG_TARGET_IS_OSX)
     endif()
 
     vcpkg_configure_make(
-        SOURCE_PATH ${SOURCE_PATH}
+        SOURCE_PATH "${SOURCE_PATH}"
         ${AUTOCONFIG}
         OPTIONS ${OPTIONS}
     )
@@ -49,22 +43,11 @@ if(VCPKG_TARGET_IS_LINUX OR VCPKG_TARGET_IS_OSX)
     vcpkg_install_make()
 
     if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "debug")
-        file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
-        file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/share)
+        file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+        file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
     endif()
 
-    configure_file(${SOURCE_PATH}/COPYING ${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright COPYONLY)
-
 elseif(VCPKG_TARGET_IS_WINDOWS)
-    vcpkg_from_github(
-        OUT_SOURCE_PATH SOURCE_PATH
-        REPO wbhart/mpir
-        REF cdd444aedfcbb190f00328526ef278428702d56e # 3.0.0
-        SHA512 f46e45bdba27c9f89953ba23186b694486fd3010bd370ea2de71a4649a2816e716a6520c9baa96936f1884437ef03f92b21c0b1fb5b757beba5a05fed30b2bfc
-        HEAD_REF master
-        PATCHES enable-runtimelibrary-toggle.patch
-    )
-
     if(VCPKG_PLATFORM_TOOLSET MATCHES "v141")
         set(MSVC_VERSION 15)
     else()
@@ -83,69 +66,69 @@ elseif(VCPKG_TARGET_IS_WINDOWS)
         set(RuntimeLibraryExt "DLL")
     endif()
 
-    file(REMOVE_RECURSE ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET})
-    file(MAKE_DIRECTORY ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET})
-    file(GLOB FILES ${SOURCE_PATH}/*)
-    file(COPY ${FILES} DESTINATION ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET})
+    file(REMOVE_RECURSE "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}")
+    file(MAKE_DIRECTORY "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}")
+    file(GLOB FILES "${SOURCE_PATH}/*")
+    file(COPY ${FILES} DESTINATION "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}")
 
     vcpkg_build_msbuild(
-        PROJECT_PATH ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/build.vc${MSVC_VERSION}/${DLL_OR_LIB}_mpir_gc/${DLL_OR_LIB}_mpir_gc.vcxproj
+        PROJECT_PATH "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/build.vc${MSVC_VERSION}/${DLL_OR_LIB}_mpir_gc/${DLL_OR_LIB}_mpir_gc.vcxproj"
         OPTIONS_DEBUG "/p:RuntimeLibrary=MultiThreadedDebug${RuntimeLibraryExt}"
         OPTIONS_RELEASE "/p:RuntimeLibrary=MultiThreaded${RuntimeLibraryExt}"
     )
 
     if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
         vcpkg_build_msbuild(
-            PROJECT_PATH ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/build.vc${MSVC_VERSION}/${DLL_OR_LIB}_mpir_cxx/${DLL_OR_LIB}_mpir_cxx.vcxproj
+            PROJECT_PATH "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/build.vc${MSVC_VERSION}/${DLL_OR_LIB}_mpir_cxx/${DLL_OR_LIB}_mpir_cxx.vcxproj"
             OPTIONS_DEBUG "/p:RuntimeLibrary=MultiThreadedDebug${RuntimeLibraryExt}"
             OPTIONS_RELEASE "/p:RuntimeLibrary=MultiThreaded${RuntimeLibraryExt}"
         )
         if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "release")
-            file(GLOB REL_LIBS_CXX ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/*/*/Release/mpirxx.lib)
+            file(GLOB REL_LIBS_CXX "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/*/*/Release/mpirxx.lib")
         endif()
         if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "debug")
-            file(GLOB DBG_LIBS_CXX ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/*/*/Debug/mpirxx.lib)
+            file(GLOB DBG_LIBS_CXX "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/*/*/Debug/mpirxx.lib")
         endif()
     endif()
 
     if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "release")
         file(GLOB HEADERS
-            ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/*/*/Release/gmp.h
-            ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/*/*/Release/gmpxx.h
-            ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/*/*/Release/mpir.h
-            ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/*/*/Release/mpirxx.h
+            "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/*/*/Release/gmp.h"
+            "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/*/*/Release/gmpxx.h"
+            "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/*/*/Release/mpir.h"
+            "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/*/*/Release/mpirxx.h"
         )
-        file(INSTALL ${HEADERS} DESTINATION ${CURRENT_PACKAGES_DIR}/include)
+        file(INSTALL ${HEADERS} DESTINATION "${CURRENT_PACKAGES_DIR}/include")
 
-        file(GLOB REL_DLLS ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/*/*/Release/mpir.dll)
-        file(GLOB REL_LIBS ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/*/*/Release/mpir.lib)
-        list(APPEND REL_LIBS  ${REL_LIBS_CXX})
+        file(GLOB REL_DLLS "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/*/*/Release/mpir.dll")
+        file(GLOB REL_LIBS "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/*/*/Release/mpir.lib")
+        list(APPEND REL_LIBS ${REL_LIBS_CXX})
 
-        file(INSTALL ${REL_DLLS} DESTINATION ${CURRENT_PACKAGES_DIR}/bin)
-        file(INSTALL ${REL_LIBS} DESTINATION ${CURRENT_PACKAGES_DIR}/lib)
+        file(INSTALL ${REL_DLLS} DESTINATION "${CURRENT_PACKAGES_DIR}/bin")
+        file(INSTALL ${REL_LIBS} DESTINATION "${CURRENT_PACKAGES_DIR}/lib")
     endif()
     if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "debug")
         file(GLOB HEADERS
-            ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/*/*/Debug/gmp.h
-            ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/*/*/Debug/gmpxx.h
-            ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/*/*/Debug/mpir.h
-            ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/*/*/Debug/mpirxx.h
+            "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/*/*/Debug/gmp.h"
+            "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/*/*/Debug/gmpxx.h"
+            "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/*/*/Debug/mpir.h"
+            "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/*/*/Debug/mpirxx.h"
         )
-        file(INSTALL ${HEADERS} DESTINATION ${CURRENT_PACKAGES_DIR}/include)
+        file(INSTALL ${HEADERS} DESTINATION "${CURRENT_PACKAGES_DIR}/include")
 
-        file(GLOB DBG_DLLS ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/*/*/Debug/mpir.dll)
-        file(GLOB DBG_LIBS ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/*/*/Debug/mpir.lib)
+        file(GLOB DBG_DLLS "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/*/*/Debug/mpir.dll")
+        file(GLOB DBG_LIBS "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/*/*/Debug/mpir.lib")
         list(APPEND DBG_LIBS  ${DBG_LIBS_CXX})
 
-        file(INSTALL ${DBG_DLLS} DESTINATION ${CURRENT_PACKAGES_DIR}/debug/bin)
-        file(INSTALL ${DBG_LIBS} DESTINATION ${CURRENT_PACKAGES_DIR}/debug/lib)
+        file(INSTALL ${DBG_DLLS} DESTINATION "${CURRENT_PACKAGES_DIR}/debug/bin")
+        file(INSTALL ${DBG_LIBS} DESTINATION "${CURRENT_PACKAGES_DIR}/debug/lib")
 
         if(VCPKG_LIBRARY_LINKAGE STREQUAL static)
-            file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/bin ${CURRENT_PACKAGES_DIR}/debug/bin)
+            file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
         endif()
     endif()
 
     vcpkg_copy_pdbs()
-
-    file(INSTALL ${SOURCE_PATH}/COPYING.lib DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
 endif()
+
+file(INSTALL "${SOURCE_PATH}/COPYING" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)

--- a/ports/mpir/vcpkg.json
+++ b/ports/mpir/vcpkg.json
@@ -1,7 +1,6 @@
 {
   "name": "mpir",
-  "version": "3.0.0",
-  "port-version": 12,
+  "version-date": "2022-03-02",
   "description": "Multiple Precision Integers and Rationals",
   "homepage": "https://github.com/wbhart/mpir",
   "license": "GPL-3.0-only",

--- a/ports/mpir/vcpkg.json
+++ b/ports/mpir/vcpkg.json
@@ -4,5 +4,6 @@
   "port-version": 12,
   "description": "Multiple Precision Integers and Rationals",
   "homepage": "https://github.com/wbhart/mpir",
+  "license": "GPL-3.0-only",
   "supports": "!(uwp | arm)"
 }

--- a/ports/mpir/vcpkg.json
+++ b/ports/mpir/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "mpir",
   "version": "3.0.0",
-  "port-version": 11,
+  "port-version": 12,
   "description": "Multiple Precision Integers and Rationals",
   "homepage": "https://github.com/wbhart/mpir",
   "supports": "!(uwp | arm)"

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -4577,8 +4577,8 @@
       "port-version": 2
     },
     "mpir": {
-      "baseline": "3.0.0",
-      "port-version": 12
+      "baseline": "2022-03-02",
+      "port-version": 0
     },
     "mpmcqueue": {
       "baseline": "2019-07-26",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -4578,7 +4578,7 @@
     },
     "mpir": {
       "baseline": "3.0.0",
-      "port-version": 11
+      "port-version": 12
     },
     "mpmcqueue": {
       "baseline": "2019-07-26",

--- a/versions/m-/mpir.json
+++ b/versions/m-/mpir.json
@@ -1,9 +1,9 @@
 {
   "versions": [
     {
-      "git-tree": "2260bc1331a81e10b4577347f0ef293b54c4da23",
-      "version": "3.0.0",
-      "port-version": 12
+      "git-tree": "5358d4a724061eab499969ae3b56f8abbdea3347",
+      "version-date": "2022-03-02",
+      "port-version": 0
     },
     {
       "git-tree": "8aabcec72d2006d0a338159fac489aae1da57220",

--- a/versions/m-/mpir.json
+++ b/versions/m-/mpir.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "b2f8170e987eadd5296a0cb187e0de2b4e866ef7",
+      "version": "3.0.0",
+      "port-version": 12
+    },
+    {
       "git-tree": "8aabcec72d2006d0a338159fac489aae1da57220",
       "version": "3.0.0",
       "port-version": 11

--- a/versions/m-/mpir.json
+++ b/versions/m-/mpir.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "b2f8170e987eadd5296a0cb187e0de2b4e866ef7",
+      "git-tree": "2260bc1331a81e10b4577347f0ef293b54c4da23",
       "version": "3.0.0",
       "port-version": 12
     },

